Classes/Link.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sitegeist\Archaeopteryx;
+
+use GuzzleHttp\Psr7\Uri;
+use Neos\Flow\Annotations as Flow;
+use Psr\Http\Message\UriInterface;
+
+/**
+ * Link value object modeled partially after the html spec for <a> tags.
+ *
+ * This Link can be used also as node property:
+ *
+ * ```yaml
+ * My.Content:
+ *   properties:
+ *     link:
+ *       type: 'Sitegeist\Archaeopteryx\Link'
+ *       ui:
+ *         inspector:
+ *           editorOptions:
+ *             anchor: true
+ *             title: true
+ *             targetBlank: true
+ *             relNofollow: true
+ * ```
+ *
+ * Note that currently the link editor can only handle and write to
+ * the property {@see Link::$target} "_blank" | null
+ * and to {@see Link::$rel} ["noopener"] | null
+ *
+ * The Link values can be accessed in Fusion the following:
+ *
+ * ```fusion
+ * href = ${q(node).property("link").href}
+ * title = ${q(node).property("link").title}
+ * # ...
+ * ```
+ *
+ * In case you need to cast the uri in {@see Link::$href} explicitly to a string
+ * you can use: `String.toString(link.href)`
+ *
+ * @Flow\Proxy(false)
+ */
+final class Link implements \JsonSerializable
```

Comment:
In general I dislike implementing \Stringable for anything that does not represent a string and only a string. Also, supporting 1:1 replacement capabilities feels a bit like miscommunication (as this is far more than a 1:1 replacement). On the other hand: With virtually no static analysis tools in standard Fusion, \Stringable will most likely save users from a lot of PITA. So a reluctant +1 from me for \Stringable

Reply:
I agree with Bernhard especially to not implement __toString in value objects when they are used in php. We lately had the same discussion in our beloved escr if the nodeAggregateId VO should implement \Stringable, and we decided against this as type coercion in php would allow to pass this nodeAggregateId to methods which accept strings which is generally probably an accident.
I build this link mostly in mind for presentation objects, but surely i would love to see adaption in fusion. I agree implementing this __toString by simply returning magically the href (and omitting the rest) would make adaption at first easier, but with neos 9 on the horizon it seems we have to start dealing with real, unforgivable, value objects in fusion as well and might as well start now?

Reply:
side note: string type coercion is not an issue if you declare(strict_types=1); , so that won't be a problem in proper PHP.

Reply:
Just as a note. I now consider it a safe default to start with not implementing Stringable at least in the beginning and see where it leads us.